day3/README.md

```diff
@@ -44,13 +44,97 @@ React.render(<Game data={ data } />, document.body);
 
-This was just an example of one child component, but there are now problem with React to create as many children as you want. 
+This was just an example of one child component, but there are now problem with React to create as many children as you want.
 
-### Redux
-store, reducer, action (objekt med evt type), dispatch, rerendring billig med React, devtools!, objkt assign
+## Redux
+All applications needs a way to handle state (server response, cached data, selected button or filter etc.). And as an app grows larger and becomes more complex, a simple way to handle the state is essential.
 
+Redux is a predictable state container which means that, by imposing restriction on how and when updates can happen, you can be sure that you always know what your state will be like.
 
-# Unidirectional data flow
+So let´s dig deeper into the most important concepts of Redux.
+
+//store, reducer, action (objekt med evt type), dispatch, rerendring billig med React, devtools!, objkt assign
+
+### Actions
+Actions are the way to pass data in Redux. It's just plain JavaScript objects that consist of type, and if you want, you can also pass data payload with the action.
+
+Example
+```javascript
+const GAME_RESET = 'GAME_RESET'
+{
+  type: GAME_RESET
+}
+```
+
+#### Action creator
+Action creators are functions that create actions. To trigger an action, you'll need to run the dispatch function with the action.
+
+Example
+```javascript
+function resetGame(index) {
+  return {
+    type: GAME_RESET
+  }
+}
+
+dispatch(resetGame())
+```
+
+### Reducers
+Actions describes that something happens and that's all the actions do. We need something that can change the state and do an act based on the action. This is where the reducers will do their magic.
+
+The reducer is pure, which means that it takes the previous state and the action, and creates the next state.
+```
+(previousState, action) => newState
+```
+Do you remember that we used reducers in day 2? https://github.com/bekk/funksjonell-frontend/tree/master/day2#reduce
+Back then we used it to calculate a sum of Oscars. In Redux we are using reducers to update the state and store it as one JavaScript object.
+
+Here is one an example of a reducer that we will work with today.
+
+```javascript
+
+const gameReducer = (state = initialStateGame, action) => {
+  switch (action.type) {
+  case CARD_FLIPPED:
+    return Object.assign({}, state, {
+      rounds: state.rounds + 1
+    });
+  default:
+    return state;
+  }
+};
+
+```
+
+#### Object.assign() - Part of ES6
+As you can see in the reducer, we are using a function called Object.assign. Let's have a look and see what it does.
+
+To ensure that the current state is not mutated, Object.assign copies the values from one source object to a target object.
+
+```javascript
+var state = { rounds: 0 };
+var action = { rounds: state.rounds + 1 }
+
+var newState = Object.assign({}, state, action);
+
+//console.log(newState) ==> { rounds: 1 }
```

Comment:
Er det mulig å heller bruke et objekt med to props? Da synes jeg det kommer enda bedre frem.